subsys/net/ip/tcp2.c

```diff
@@ -635,16 +779,19 @@ static struct tcp *tcp_conn_alloc(void)
 
 	conn->state = TCP_LISTEN;
 
-	conn->win = tcp_window;
+	conn->recv_win = tcp_window;
 
 	sys_slist_init(&conn->send_queue);
 
 	k_delayed_work_init(&conn->send_timer, tcp_send_process);
-
+	k_delayed_work_init(&conn->data_timer, tcp_resend_queued_data);
 	k_delayed_work_init(&conn->timewait_timer, tcp_timewait_timeout);
 
 	tcp_conn_ref(conn);
 
+	conn->seq = tcp_init_isn();
+	conn->last_seq = conn->seq + 1;
```

Comment:
The last_seq is confusing and perhaps need more thought how to eliminate it.

Reply:
Yes, that might need some tweaking. Will investigate but currently the plan is to get this thing working by 29.5 so we can get things working ok in 2.3. All the refactoring can be postponed after that.

Reply:
I propose to just use conn->seq and conn->unacked_len for these calculations and drop last_seq.